### PR TITLE
BUG: Fix getsizeof when using Series(obj) and taking into account GC corrections

### LINKED ISSUE #51858: BUG: Get sizeof fails for pandas series or data frame when one element is a type
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import sys
df=pd.Series(str)

sys.getsizeof(df)
TypeError: descriptor '__sizeof__' of 'str' object needs an argument
```

### Issue Description

Python getsizeof fails on a series which holds a type as data.

More details in the SO question and answer
https://stackoverflow.com/questions/75682987/why-does-sys-getsizeof-fail-on-a-pandas-series-or-data-frame-when-they-hold-a-ty/75683341#75683341

### Expected Behavior

No type error

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 2e218d10984e9919f0296931d92ea851c6a6faf5
python           : 3.9.16.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.10.147+
Version          : #1 SMP Sat Dec 10 16:00:40 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.3
numpy            : 1.22.4
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 57.4.0
pip              : 22.0.4
Cython           : 0.29.33
pytest           : 3.6.4
hypothesis       : None
sphinx           : 3.5.4
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.2
html5lib         : 1.0.1
pymysql          : None
psycopg2         : 2.9.5
jinja2           : 3.1.2
IPython          : 7.9.0
pandas_datareader: 0.9.0
bs4              : 4.6.3
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2023.3.0
gcsfs            : None
matplotlib       : 3.5.3
numba            : 0.56.4
numexpr          : 2.8.4
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : 0.17.9
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.1
snappy           : None
sqlalchemy       : 1.4.46
tables           : 3.7.0
tabulate         : 0.8.10
xarray           : 2022.12.0
xlrd             : 1.2.0
xlwt             : 1.3.0
zstandard        : None
tzdata           : 2022.7

</details>

### PR DESCRIPTION
**Pull request by balexandermunoz**
Calling __sizeof__ to compute the size of an object produces an error, and in general, this method does not take Garbage Collector corrections into account.

- [x] closes #51858
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/base/test_sizeof.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from sys import getsizeof
+
+from pandas import (
+    DataFrame,
+    Series,
+)
+
+
+def test_sysof():
+    getsizeof(DataFrame)
```
Comment (mroeschke) [inline pandas/tests/base/test_sizeof.py:10]: Could you `assert` the result of all these calls?

**Comment by mroeschke** [comment]
Could you put these tests in `pandas/tests/base/test_misc.py`

File: pandas/tests/base/test_sizeof.py (hunk 1 above)
Comment (balexandermunoz) [reply pandas/tests/base/test_sizeof.py]: I just used one of the existing tests (test_memory_usage) by simply adding more Series.
Thank you very much. 
I will keep an eye out if anything else is required.

File: pandas/conftest.py [hunk 2]
```diff
@@ -760,6 +760,29 @@ def index_or_series_obj(request):
     return _index_or_series_objs[request.param].copy(deep=True)
 
 
+_object_series = {
```
Comment (mroeschke) [inline pandas/conftest.py:763]: Nit: Could you rename this to `_typ_objects_series`?

Comment (balexandermunoz) [reply pandas/conftest.py]: Done!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @balexandermunoz

**Pull request merged**